src/core/config.ts

```
@@ -176,120 +207,109 @@ export async function updateConfig(cwd: string, key: string, value: unknown): Pr
 
-  if (parsed.version === '1.0') {
-    return true;
+  if (parsed.syncMode && parsed.syncMode !== 'native') {
+    throw new Error(UNSUPPORTED_SYNC_MODE_MESSAGE);
   }
 
-  const needsSkillDirectoryMigration =
-    Array.isArray(parsed.skillDirectories) &&
-    parsed.skillDirectories.some((dir) => typeof dir === 'string');
+  return {
+    version: parsed.version === '1.0' ? '2.0' : parsed.version ?? '2.0',
```

ESLint

[error] Replace `parsed.version·??·'2.0'` with `(parsed.version·??·'2.0')`

(prettier/prettier)